WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -49,12 +49,27 @@ class PostCoordinator: NSObject {
                 }
                 switch state {
                 case .ended:
-                    self.updateReferences(to: media, in: post)
-
-                    // Let's check if media uploading is still going, if all finished with success then we can upload the post
-                    if !self.mediaCoordinator.isUploadingMedia(for: post) && !post.hasFailedMedia {
-                        self.removeObserver(for: post)
-                        self.upload(post: post)
+                    let successHandler = {
+                        self.updateReferences(to: media, in: post)
+                        // Let's check if media uploading is still going, if all finished with success then we can upload the post
+                        if !self.mediaCoordinator.isUploadingMedia(for: post) && !post.hasFailedMedia {
+                            self.removeObserver(for: post)
+                            self.upload(post: post)
+                        }
+                    }
+                    switch media.mediaType {
+                    case .video:
+                        EditorMediaUtility.fetchRemoteVideoURL(for: media, in: post) { [weak self] (result) in
```

Comment:
@SergioEstevao I have second thoughts about where to put this service call because this is being called for classical editor posts also. is there any chance you know sth better to put this?